crates/core/src/backend/decrypt.rs

```diff
@@ -395,12 +411,41 @@ impl<C: CryptoKey> DecryptWriteBackend for DecryptBackend<C> {
             }
             None => self.key().encrypt_data(data)?,
         };
-        let id = hash(&data);
-        self.write_bytes(tpe, &id, false, data.into())
+        if self.extra_check {
```

Comment:
If we do it as a default, do we want to branch here, or would we rather branch when integrity_check is not set?

Reply:
actually I think we should prefer clean code to premature optimization. (I doubt that branch prediction will make any measurable difference here)